index.bs

```diff
@@ -361,7 +361,64 @@ Report long tasks {#report-long-tasks}
 
-Long Tasks API adheres to the same-origin policy by including origin-safe attribution information about the source of the long task.
-There is a 50ms threshold for long tasks. Together this provides adequate protection against security attacks against browser.
+Long Tasks API adheres to the same-origin policy by including origin-safe attribution information about
+the source of the long task. There is a 50ms threshold for long tasks. Together this provides adequate
+protection against security attacks against cross-origin leaks.
+
+The Long Tasks API provides timing information about the duration and type of tasks executed by the user,
+as well as attribution such as source code location for function calls. This could enable an attacker
+to perform side-channel timing attacks to guess user’s action, or identify the user. For example, a pattern of
+long script followed by a long render could be put together to guess user’s interaction with a social
+widget. Detailed function call attribution would be used to determine the user’s action.
+
+While the API doesn’t introduce any new privacy attacks, it could make existing privacy attacks faster.
+Mitigations for this are possible and can be implemented as needed:
+
+* Further clamp the long task duration provided by the API to make attacks harder to exploit.
+* Limit the number of origins for which longtasks are exposed by the API, and obfuscate the attribution of
+    any tasks afterwards.
+* Allow dropping the culprit/attribution information after a certain threshold.
+* Add a built-in delay to the timing information exposed to make attacks dependent on longtask volume harder
+    to execute.
+
+What is Exposed to Observers? {#what-is-exposed}
+--------------------------------------------------------
+
+All observers within the top level page (i.e. all iframes in the page and the main frame) will receive
+notifications about presence of long tasks. We expose the start time of the task, its duration, and a
+pointer to the culprit frame. This information can already be observed today, and with higher resolution,
+using setTimeout. This can be done by an attacker by clearing everything else on the page and adding the
+vulnerable cross-origin resource to ensure that delays from the setTimeout are caused by that resource.
+Observers in other different pages (tabs or windows) should not receive notifications, regardless of the
+architecture of the user agent.
+
+Cross origin rules for what is exposed:
+* Cross-origin observers may see the direction of the culprit e.g if the culprit is a deeply nested iframe,
+    then the host page can see the first cross-origin between itself and the culprit.
+* Conversely, if the culprit is the top level page, then a deeply embedded iframe can see that a longtask
+    occurrred in its cross-origin ancestor but does not receive any information about it.
+
+Opt-in: In a future version of this API, sites could opt into revealing their attribution info to trusted
```

Comment:
I don't know if we should discuss attribution much as it's not currently implemented nor do we know what the opt-in would look like (and if we can avoid it in some cases)

Reply:
Ok, for some reason that was already in the doc, but sure we can remove it until we add such opt-in.